tests/unit/test_topictags.py

```diff
@@ -0,0 +1,538 @@
+# Copyright (c) 2015 Amazon.com, Inc. or its affiliates.  All Rights Reserved
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the
+# "Software"), to deal in the Software without restriction, including
+# without limitation the rights to use, copy, modify, merge, publish, dis-
+# tribute, sublicense, and/or sell copies of the Software, and to permit
+# persons to whom the Software is furnished to do so, subject to the fol-
+# lowing conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+# OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABIL-
+# ITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT
+# SHALL THE AUTHOR BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+# WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+# IN THE SOFTWARE.
+#
+import json
+
+from nose.tools import assert_equal
+import mock
+
+from awscli.testutils import unittest, FileCreator
+from awscli.topictags import TopicTagDB
+from awscli.compat import six
+
+
+class TestTopicTagDB(unittest.TestCase):
+    def setUp(self):
+        self.topic_tag_db = TopicTagDB()
+        self.file_creator = FileCreator()
+
+        # In python 3, order matters when calling assertEqual to
+        # compare lists and dictionaries with lists. Therefore,
+        # assertItemsEqual needs to be used but it is renamed to
+        # assertCountEqual in python 3.
+        if six.PY3:
+            self.assertItemsEqual = self.assertCountEqual
```

Comment:
It doesn't actually seem like assertCountEqual is necessary.  The lists you're comparing don't contain dupes do they?  You can either just have a custom assertion method that just `set()`'s the values before `assertEqual` or `sorted()'d` the data before calling assertEqual.

Either way I'm not a fan of these version conditionals throughout the code and prefer to have them all collected to a single location wherever possible.  If you'd prefer to stick with assertCountEqual, I'd suggest moving these differences into `awscli.testutils` which is where all the various python version differences related to testing are encapsulated.